bolts-tasks/src/test/java/bolts/TaskTest.java

```diff
@@ -311,6 +312,27 @@ public Integer call() throws Exception {
   }
 
   @Test
+  public void testBackgroundCallWaitingWithTimeouts() throws Exception {
+    final Object sync = new Object();
+
+    Task<Integer> task = Task.callInBackground(new Callable<Integer>() {
+      public Integer call() throws Exception {
+        synchronized (sync) {
+          sync.wait();
+        }
+        return 5;
+      }
+    });
+    assertFalse(task.waitForCompletion(100, TimeUnit.MILLISECONDS));
+    synchronized (sync) {
+      sync.notify();
+    }
+    task.waitForCompletion();
+    assertTrue(task.waitForCompletion(100, TimeUnit.MILLISECONDS));
+    assertEquals(5, task.getResult().intValue());
+  }
```

Comment:
It looks like we'd want to verify the third code path where task.waitForCompletion returns true if it isn't already completed when calling it.